android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/PwoDiscoveryService.java

```diff
@@ -117,12 +139,11 @@ private void initialize() {
     for (PwoDiscoverer pwoDiscoverer : mPwoDiscoverers) {
       pwoDiscoverer.setCallback(this);
     }
+    mPwoResponseCallbacks = new ArrayList<>();
```

Comment:
Unless I'm misunderstanding you, I'm keeping each variable together.  All the manipulation of mPwoDiscoverers is one block.